pkg/controllers/route_controller.go

```diff
@@ -456,3 +430,196 @@ func (r *routeReconciler) validateBackendRefsIpFamilies(ctx context.Context, rou
 
 	return nil
 }
+
+var (
+	ErrValidation          = errors.New("validation")
+	ErrParentRefsNotFound  = errors.New("parentRefs are not found")
+	ErrRouteGKNotSupported = errors.New("route GroupKind is not supported")
+)
+
+// Validation for route spec. Will validate and update route status. Returns error if not valid.
+// Validation rules are suppose to be compliant to Gateway API Spec.
+//
+//	https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io%2fv1.RouteConditionType
+//
+// There are 3 condition types: Accepted, PartiallyInvalid, ResolvedRefs.
+// We dont support PartiallyInvalid for now and reject entire route if there is at least one invalid field.
+// Accepted type is related to parentRefs, and ResolvedRefs to backendRefs. These 2 are validated independently.
+func (r *routeReconciler) validateRoute(ctx context.Context, route core.Route) error {
```

Comment:
Variable names in this function are hard to understand. Let's please stick to camelCase words

Reply:
updated